test/shared/KestrelTestLoggerProvider.cs

```diff
@@ -11,7 +11,7 @@ public class KestrelTestLoggerProvider : ILoggerProvider
         private readonly ILogger _testLogger;
 
         public KestrelTestLoggerProvider()
-            : this(new TestApplicationErrorLogger())
+            : this(new TestApplicationErrorLogger() { ThrowOnCriticalErrors = false })
```

Comment:
Don't do this for all tests For tests where you are expecting a critical errors to be logged, I would just call `new KestrelTestLoggerProvider(new TestApplicationErrorLogger() { ThrowOnCriticalErrors = false })` explicitly.

The idea is that most tests should cause critical errors to be logged, so having this can help us catch issues that might have otherwise been missed.